new/rfcXXX-distro-sugar-helpers.md

```diff
@@ -0,0 +1,159 @@
+---
+RFC: unassigned
+Title: My Title
+Author: Alan Smithee <asmithee@example.com>
+Status: Draft
+Type: <Standards Track, Informational, Process>
+<Replaces: RFCxxx>
+<Tracking:>
+<  - https://github.com/chef/chef/issues/X>
+---
+
+# Title
+
+Merging chef-sugar platform and `platform_family` helpers, along with higher level helpers.
```

Comment:
i really hate the attributes method_missing syntax for one, that's a major :-1: because anything method_missing needs to die in a fire.